Performance improvements for 'blip' and 'viz' docker services.

cbwebdevelopment opened the pull request:
Goes hand-in-hand with:
https://github.com/tidepool-org/viz/pull/93
https://github.com/tidepool-org/blip/pull/451

- Changes to the `docker-compose.yml` 
  - allow us to not volume-mount the `node_modules` and `dist` directories to improve local performance in the `blip` and `viz` services.
- Updates to the `tidepool` helper script
  - added `tidepool stop` command
  - added `tidepool pull` command
  - added `tidepool api` command
- Documentation updates
  - new helper script commands
  - instructions for overriding the default docker-compose.yml file with an override compose file

cbwebdevelopment commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Docker performance improvements for 'blip' and 'viz' services](https://trello.com/c/gJMqJ9K6/287-docker-performance-improvements-for-blip-and-viz-services)

pazaan commented on docker-compose.yml:
```diff
@@ -183,24 +186,54 @@ services:
     ports:
       - '9221:9221'
 
+  # tools:
```
What does the tools container do?

cbwebdevelopment replied:
Sorry about that... I should remove it.  It was to run the Platform tools API (tapi).  I had been using it a bit for quick 'dummy' account creation, and was going to leave it as placeholder until I got the tools Dockerfile merged into the `platform` repo, though, I now doubt it will happen.

I'll take this out, and remove reference to the tools service in the README.

cbwebdevelopment commented:
I've removed the references to the `tools` service, @pazaan

cbwebdevelopment commented:
Also, after cleaning up the `blip` and `viz` Dockerfiles as per your recommendations, I also added a `viz-dist` named volume, and removed the unneeded `viz-modules` volume, as I realized I don't need the `viz` node_modules in the `blip` service, only the `/dist/index.js` file to allow for proper live re-compiling when npm linking `@tidepool/viz` into `blip`
